xmr-btc/src/happy_path.rs

```diff
@@ -0,0 +1,216 @@
+//! This module shows how a BTC/XMR atomic swap proceeds along the happy path.
+
+use crate::{alice, bitcoin, bob, monero, Message, ReceiveMessage, SendMessage};
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use bitcoin_harness::Bitcoind;
+use futures::{
+    channel::{
+        mpsc,
+        mpsc::{Receiver, Sender},
+    },
+    SinkExt, StreamExt,
+};
+use monero_harness::Monero;
+use rand::rngs::OsRng;
+use std::convert::TryInto;
+use testcontainers::clients::Cli;
+
+const TEN_XMR: u64 = 10_000_000_000_000;
+
+pub async fn init_bitcoind(tc_client: &Cli) -> Bitcoind<'_> {
+    let bitcoind = Bitcoind::new(tc_client, "0.19.1").expect("failed to create bitcoind");
+    let _ = bitcoind.init(5).await;
+
+    bitcoind
+}
+
+/// Create two mock `Transport`s which mimic a peer to peer connection between
+/// two parties, allowing them to send and receive `thor::Message`s.
+pub fn make_transports() -> (Transport, Transport) {
+    let (a_sender, b_receiver) = mpsc::channel(5);
+    let (b_sender, a_receiver) = mpsc::channel(5);
+
+    let a_transport = Transport {
+        sender: a_sender,
+        receiver: a_receiver,
+    };
+
+    let b_transport = Transport {
+        sender: b_sender,
+        receiver: b_receiver,
+    };
+
+    (a_transport, b_transport)
+}
+
+#[derive(Debug)]
+pub struct Transport {
+    sender: Sender<Message>,
+    receiver: Receiver<Message>,
+}
+
+#[async_trait]
+impl SendMessage for Transport {
+    async fn send_message(&mut self, msg: Message) -> Result<()> {
+        self.sender
+            .send(msg)
+            .await
+            .map_err(|_| anyhow!("failed to send message"))
+    }
+}
+
+#[async_trait]
+impl ReceiveMessage for Transport {
+    async fn receive_message(&mut self) -> Result<Message> {
+        let msg = self
+            .receiver
+            .next()
+            .await
+            .ok_or_else(|| anyhow!("failed to receive message"))?;
+
+        Ok(msg)
+    }
+}
```

Comment:
This seems also to be very similar to what @rishflab did in his PR #2 in https://github.com/comit-network/xmr-btc-swap/blob/b6df715b1bdef81f20333145e1705cbdcf0e4fb8/xmr-btc/tests/harness/transport.rs#L16-L55

The difference is that @rishflab made a difference in the message types and uses two different enums, one for Alice's messages and one for Bob's messages. This should prevent anyone from sending the wrong message.